### PR TITLE
feat(table): resizable columns

### LINKED ISSUE #496: Resizable table columns
**Describe the problem this feature request is solving**
This is something we uncovered in our HAS user testing that I hadn't previously thought of.  In the Atlas test, the user was looking for a field called "HospitalAccount".  The data set had numerous instances of "HospitalAccount2", "HospitalAccount3", etc.  But because of the width of the column, the content was truncated before the number.  So it was a pretty serious usability issue not to be able to see a wider column.

**Describe the solution you'd like**
For our table component, we should allow users to resize table columns on their own so they can control how much content they see.  This is a good example the functionality I'd be looking for: https://codepen.io/validide/pen/aOKLNo

**Describe alternatives you've considered**
We have tooltips on the truncated cells, but most users missed that, or said it would be annoying to have to do that for every row to try and scan the data.

**If any, which components will have to change, and how**
hc-table; shouldn't be a breaking change

### PR DESCRIPTION
**Pull request by andrew-frueh**
I've been occasionally checking in on [this CDK forum thread](angular/material2#8312) to see if any progress had been made.  And it turned out that one developer posted their own solution, which was enough to get me started.  Ended up modifying his approach a decent bit to fit our library better, his stuff provided the foundation so I linked to it where appropriate in the code comments.

Check the usage markdown explanation for some details on the way I approached this.  I'm sure there are ways to tune this up further, particularly when tables are scaled.  The developer's code that I started from did use two open source utility functions.  One I added to our utils.ts file, and the other I just kept in the table component.  They seemed compact enough so as not to bloat anything in the library, but let me know if you think otherwise.

Excited to get this functionality added because it's come up as a usability issue in some of our user tests.  So thanks in advance for any input you have on this PR!

closes #496

### CODE REVIEW

File: projects/cashmere/src/lib/table/bind-observable/bind-observable-options.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    key?: string;
+    /**
+     * If true, the observable will emit the raw setter value if the decorated property is an accessor.
+     * Otherwise the observable will call the getter once and emit the returned value.
+     * This option will be ignored if the decorated property has no get accessor.
+     */
+    emitRawSetterValue?: boolean;
+}
+
+export function isBindObservableOpts(pet: any): pet is BindObservableOpts {
```
Comment (jonest) [inline projects/cashmere/src/lib/table/bind-observable/bind-observable-options.ts]: Just curious what lead you to choose `pet` as the variable name. Seems kind of funky haha!

File: projects/cashmere/src/lib/table/bind-observable/bind-observable.spec.ts [hunk 2]
```diff
@@ -0,0 +1,271 @@
+import {BindObservable} from './bind-observable';
+import {Observable} from 'rxjs';
+import {take} from 'rxjs/operators';
+
+function WrappingDecorator(): PropertyDecorator {
+    return (target: any, propertyName: string | symbol) => {
```
Comment (jonest) [inline projects/cashmere/src/lib/table/bind-observable/bind-observable.spec.ts]: I have no idea what a `symbol` type is even after looking at the docs. How is it used exactly?

File: projects/cashmere/src/lib/table/cell-resizer.component.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+export class CellResizeEvent {
+    constructor(public width: number, public directionLeft: boolean) {}
+}
+
+@Component({
+    selector: 'hc-cell-resizer',
+    template: '<div class="hc-cell-resizer-left"></div><div class="hc-cell-resizer-right"></div>',
+    styleUrls: ['./cell-resizer.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
```
Comment (jonest) [inline projects/cashmere/src/lib/table/cell-resizer.component.ts]: If anyone else is unfamiliar with `ChangeDetectionStrategy.OnPush`, this post helped me: https://blog.thoughtram.io/angular/2016/02/22/angular-2-change-detection-explained.html

**Comment by andrew-frueh** [comment]
Hey @jonest - I was just on BrowserStack and noticed the same on IE11 :-(  I'll see if I can get to the bottom of that this afternoon.

**Comment by jonest** [comment]
> Hey @jonest - I was just on BrowserStack and noticed the same on IE11 :-( I'll see if I can get to the bottom of that this afternoon.

You probably looked already, but I didn't see anything obvious like a console error or anything. Good luck! 🍀 🤞

**Comment by jonest** [comment]
Ping me on Slack if you push up some changes!

File: projects/cashmere/src/lib/table/bind-observable/bind-observable-options.ts (hunk 1 above)
Comment (andrew-frueh) [reply projects/cashmere/src/lib/table/bind-observable/bind-observable-options.ts]: Hah - afraid I can't claim to know much of anything when it comes to the bind observable functions, they're an open source utility set that the person who originally posted their column resizing solution was using:

https://github.com/PSanetra/bind-observable

So who knows what they were thinking when they named their variables :-)

Comment (jonest) [reply projects/cashmere/src/lib/table/bind-observable/bind-observable-options.ts]: Got it! I didn't see any source attribution so I assumed it was something you wrote. Think we should include that there?

Comment (andrew-frueh) [reply projects/cashmere/src/lib/table/bind-observable/bind-observable-options.ts]: Yup - sounds good.  I put the copyright and license at the top of the file, but I'll attribute it as well.

**Comment by andrew-frueh** [comment]
Thanks as always for the input @jonest - I just added a commit that updates the attributions for the Open Source stuff so it's in every file and links to the GitHub project.

Also tracked down the IE11 bug.  Turns out the IE doesn't support `MouseEvent.movementX` so you have to calculate the equivalent yourself (storing the previous value):

`currentEvent.movementX = currentEvent.screenX - previousEvent.screenX`

**Comment by jonest** [comment]
> Thanks as always for the input @jonest - I just added a commit that updates the attributions for the Open Source stuff so it's in every file and links to the GitHub project.
> 
> Also tracked down the IE11 bug. Turns out the IE doesn't support `MouseEvent.movementX` so you have to calculate the equivalent yourself (storing the previous value):
> 
> `currentEvent.movementX = currentEvent.screenX - previousEvent.screenX`

Nice work! I'll check out the changes now to verify 😃

**Pull request merged**